src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -4849,53 +4850,60 @@ def work(self, cu):
+                    self._log.info('Operating from staging')
+                    # Remove the leading slash to get a relative path from the staging area
+                    rel2staging = source_url.path.split('/',1)[1]
+                    source = os.path.join(staging_area, rel2staging)
+                else:
+                    self._log.info('Operating from absolute path')
+                    source = source_url.path
 
-            # Get the target from the directive and convert it to the location
-            # in the workdir
-            target = directive['target']
-            abs_target = os.path.join(workdir, target)
+                # Get the target from the directive and convert it to the location
+                # in the workdir
+                target = directive['target']
+                abs_target = os.path.join(workdir, target)
 
-            # Create output directory in case it doesn't exist yet
-            rec_makedir(os.path.dirname(abs_target))
+                # Create output directory in case it doesn't exist yet
+                rec_makedir(os.path.dirname(abs_target))
 
-            self._log.info("Going to '%s' %s to %s", directive['action'], source, abs_target)
+                self._log.info("Going to '%s' %s to %s", directive['action'], source, abs_target)
 
-            if   directive['action'] == LINK: os.symlink     (source, abs_target)
-            elif directive['action'] == COPY: shutil.copyfile(source, abs_target)
-            elif directive['action'] == MOVE: shutil.move    (source, abs_target)
-            else:
-                # FIXME: implement TRANSFER mode
-                raise NotImplementedError('Action %s not supported' % directive['action'])
+                if   directive['action'] == LINK: os.symlink     (source, abs_target)
+                elif directive['action'] == COPY: shutil.copyfile(source, abs_target)
+                elif directive['action'] == MOVE: shutil.move    (source, abs_target)
+                else:
+                    # FIXME: implement TRANSFER mode
+                    raise NotImplementedError('Action %s not supported' % directive['action'])
 
-            log_message = "%s'ed %s to %s - success" % (directive['action'], source, abs_target)
-            self._log.info(log_message)
+                log_message = "%s'ed %s to %s - success" % (directive['action'], source, abs_target)
+                self._log.info(log_message)
 
+        except Exception as e:
+            self._log.exception("staging input failed -> unit failed")
+            staging_ok = False
 
-        self._prof.prof('log', msg="toward agent scheduling", uid=cu['_id'])
 
-      # self.advance(cu, rp.AGENT_SCHEDULING_PENDING, publish=True, push=True)
-        self.advance(cu, rp.ALLOCATING_PENDING, publish=True, push=True)
+        # Agent input staging is done (or failed)
+        if staging_ok:
+          # self.advance(cu, rp.AGENT_SCHEDULING_PENDING, publish=True, push=True)
```

Comment:
that's a preparation for the last set of pending state renames.  `ALLOCATING` is not yet in sync with the target state model.  There are some more, similar comments somewhere through the agent I believe...